### PR TITLE
Determine crpix from saif toggle

### PR DESCRIPTION
**Pull request by wbalmer**
See discussion in #114 . For data affected by bad fits writing behavior, for instance, the NIRCam BAR/NARROW offset position, it might be necessary to toggle calling the CRPIX values from pysaif, as opposed to recording them from the headers. This affects the position of the mask and therefore the coronagraphic throughput. Default behavior is unchanged and the toggle is set to false. Added to the appropriate functions in `database.py`

Potential improvement could look to incorporate @JarronL 's `webbpsf_ext.imreg_tools.get_coron_apname` function in the default database building step, but this change is non-intrusive and sufficient for current reductions (unlike the original implementation in #114 ).

### CODE REVIEW

File: spaceKLIP/database.py [hunk 1]
```diff
@@ -109,6 +109,7 @@ def read_jwst_s012_data(self,
                             datapaths,
                             psflibpaths=None,
                             bgpaths=None,
+                            cr_from_saif=False,
```
Comment (AarynnCarter) [inline spaceKLIP/database.py:112]: Fix typo for "saif" -> "siaf"

File: spaceKLIP/database.py [hunk 2]
```diff
@@ -274,8 +275,12 @@ def read_jwst_s012_data(self,
             BLURFWHM += [head.get('BLURFWHM', np.nan)]
             head = hdul['SCI'].header
             BUNIT += [head.get('BUNIT', 'NONE')]
-            CRPIX1 += [head.get('CRPIX1', np.nan)]
-            CRPIX2 += [head.get('CRPIX2', np.nan)]
+            if cr_from_saif:
```
Comment (AarynnCarter) [inline spaceKLIP/database.py:278]: Typo here too

File: spaceKLIP/database.py [hunk 3]
```diff
@@ -571,7 +576,8 @@ def read_jwst_s012_data(self,
         pass
     
     def read_jwst_s3_data(self,
-                          datapaths):
+                          datapaths,
+                          cr_from_saif=False):
```
Comment (AarynnCarter) [inline spaceKLIP/database.py:580]: Typo here too

File: spaceKLIP/database.py [hunk 4]
```diff
@@ -718,8 +724,12 @@ def read_jwst_s3_data(self,
             if TYPE[-1] == 'CORON3':
                 head = hdul['SCI'].header
             BUNIT += [head.get('BUNIT', 'NONE')]
-            CRPIX1 += [head.get('CRPIX1', np.nan)]
-            CRPIX2 += [head.get('CRPIX2', np.nan)]
+            if cr_from_saif:
```
Comment (AarynnCarter) [inline spaceKLIP/database.py:727]: Typo here too

File: spaceKLIP/database.py [hunk 5]
```diff
@@ -1360,11 +1371,13 @@ def create_database(output_dir,
         db.read_jwst_s012_data(datapaths=datapaths,
                                psflibpaths=psflibpaths,
                                bgpaths=bgpaths,
+                               cr_from_saif=cr_from_saif,
                                assoc_using_targname=assoc_using_targname)
     elif str(readlevel) == '3':
         # the above get_files usage won't match KLIP outputsa, so find them here
         datapaths_klip = sorted(glob.glob(os.path.join(input_dir, "*KLmodes-all.fits")))
         db.read_jwst_s3_data(datapaths=datapaths+datapaths_klip,
+                             cr_from_saif=cr_from_saif,
```
Comment (AarynnCarter) [inline spaceKLIP/database.py:1380]: Okay silly for me to mark them all, other than the typo looks good.

**Review by AarynnCarter — COMMENTED** [review]
Looks good @wbalmer, just some typos to fix. 

Can you confirm you've tested this on the data you're working on for both of the `cr_from_siaf` cases?

**Comment by wbalmer** [comment]
Hi Aarynn - yep! I have tested the toggle behaves well when set false by default, when set false explicitly, and when set true explicitly.

**Review by AarynnCarter — APPROVED** [review]

**Pull request merged**